### PR TITLE
Add MDM OSMOS instrument

### PR DESCRIPTION
**Pull request by profxj**
Sneaking this one in as it had been waiting and it is 
pretty simple.

Adds a new Dev suite test (on the `osmos` branch) which it passes:

--- PYPEIT DEVELOPMENT SUITE PASSED 1/1 TESTS (Masters ignored) ---

All unit tests passing:

================ 211 passed, 444 warnings in 738.52s (0:12:18) =================

### CODE REVIEW

File: pypeit/core/wavecal/templates.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def pypeit_arcspec(in_file, slit):
     wv_vac = utils.func_val(iwv_calib['fitc'], x/iwv_calib['xnorm'], iwv_calib['function'],
                            minx=iwv_calib['fmin'], maxx=iwv_calib['fmax'])
     # Return
-    return wv_vac, np.array(iwv_calib['spec'])
+    return wv_vac, np.array(iwv_calib['spec']).flatten()  # JXP added flatten on 2019-11-09
```
Comment (kbwestfall) [inline pypeit/core/wavecal/templates.py]: Can you explain why this was necessary in a comment?

File: pypeit/spectrographs/mdm_osmos.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+from pypeit.core import framematch
+from pypeit.par import pypeitpar
+from pypeit.spectrographs import spectrograph
+from pypeit.core import parse
+from pypeit.images import detector_container
+
+from IPython import embed
+
+
+class MDMOSMOSMDM4KSpectrograph(spectrograph.Spectrograph):
```
Comment (kbwestfall) [inline pypeit/spectrographs/mdm_osmos.py]: Wowza.  Try to say that 3 times fast.

File: pypeit/telescopes.py [hunk 3]
```diff
@@ -78,6 +78,13 @@ def __init__(self):
                                                latitude=32.701308,
                                                elevation=3221.0)
 
+class MDMTelescopePar(TelescopePar):
+    def __init__(self):
+        super(MDMTelescopePar, self).__init__(name='MDM',
+                                              longitude=111.616111,
+                                              latitude=31.9516666,
+                                              elevation=2098.)                # Used KPNO not MDM
```
Comment (kbwestfall) [inline pypeit/telescopes.py]: This is a bit nit picky, but, if this is really for KPNO, we should try to call this out better.  Can we instead call this `KPNOTelescope` or something, and then add a comment in the MDM spectrograph that says we're using KPNO but should change it to MDM, and then add a GitHub issue?

**Comment by kbwestfall** [comment]
PS - Did you add test data to the devsuite?

File: pypeit/core/wavecal/templates.py (hunk 1 above)
Comment (profxj) [reply pypeit/core/wavecal/templates.py]: I honestly don't recall.  It was 4 months ago.

File: pypeit/telescopes.py (hunk 3 above)
Comment (profxj) [reply pypeit/telescopes.py]: modified

**Comment by profxj** [comment]
merging

**Pull request merged**